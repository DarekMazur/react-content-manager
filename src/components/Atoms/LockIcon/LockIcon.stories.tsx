import { Meta, StoryObj } from '@storybook/react';
import LockIcon from './LockIcon.tsx';

/** Lock Icon - based on FontAwesome */
const meta: Meta<typeof LockIcon> = {
  tags: ['autodocs'],
  component: LockIcon,
  title: 'Atoms/Lock icon',
};

export default meta;

type Story = StoryObj<typeof LockIcon>;

export const Primary: Story = {};
