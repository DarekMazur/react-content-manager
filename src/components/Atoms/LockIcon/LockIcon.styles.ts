import styled from 'styled-components';

export const StyledLockIcon = styled.svg`
  font-size: 14rem;
  color: ${({ theme }) => theme.colors.darkBlue};
  margin: 3rem;
`;
