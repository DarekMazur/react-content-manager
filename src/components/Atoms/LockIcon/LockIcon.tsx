import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { StyledLockIcon } from './LockIcon.styles.ts';

const LockIcon = () => {
  return <StyledLockIcon as={FontAwesomeIcon} icon={['fas', 'lock']} />;
};

export default LockIcon;
