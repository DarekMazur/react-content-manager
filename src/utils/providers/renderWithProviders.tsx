import { render } from '@testing-library/react';
import { JSX, PropsWithChildren } from 'react';
import AppProviders from './AppProviders.tsx';

export const renderWithProviders = (
  children: PropsWithChildren<JSX.Element>,
) => {
  return render(<AppProviders>{children}</AppProviders>);
};
