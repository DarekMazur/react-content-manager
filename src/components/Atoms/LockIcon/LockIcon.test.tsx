import '@testing-library/jest-dom';
import { renderWithProviders } from '../../../utils/providers/renderWithProviders.tsx';
import LockIcon from './LockIcon.tsx';

const component = renderWithProviders(<LockIcon />);
describe('Lock Icon', () => {
  it('- rendered component match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
