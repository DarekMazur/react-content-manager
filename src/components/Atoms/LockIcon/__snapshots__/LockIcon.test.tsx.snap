// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Lock Icon > - rendered component match to snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlfmHC jOwVnv"
      >
        <div>
          <p>
            Are you sure you want to delelte 
             
            <strong />
            ? This action is permanent
          </p>
          <div>
            <button
              class="sc-bdfCDU sc-gsTDqH ebXxHs dyfnGz"
            >
              delete
            </button>
            <button
              class="sc-bdfCDU sc-gsTDqH ebXxHs jcLNlx"
            >
              cancel
            </button>
          </div>
        </div>
      </div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-lock sc-hKgKIp LCqwq"
        data-icon="lock"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M144 144v48H304V144c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192V144C80 64.5 144.5 0 224 0s144 64.5 144 144v48h16c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64H80z"
          fill="currentColor"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlfmHC jOwVnv"
    >
      <div>
        <p>
          Are you sure you want to delelte 
           
          <strong />
          ? This action is permanent
        </p>
        <div>
          <button
            class="sc-bdfCDU sc-gsTDqH ebXxHs dyfnGz"
          >
            delete
          </button>
          <button
            class="sc-bdfCDU sc-gsTDqH ebXxHs jcLNlx"
          >
            cancel
          </button>
        </div>
      </div>
    </div>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-lock sc-hKgKIp LCqwq"
      data-icon="lock"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M144 144v48H304V144c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192V144C80 64.5 144.5 0 224 0s144 64.5 144 144v48h16c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64H80z"
        fill="currentColor"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
